sdk/core/Azure.Core/src/DynamicData/MutableJsonElement.cs

```diff
@@ -1055,6 +1056,13 @@ private object GetSerializedValue(object value)
                 mje.EnsureValid();
             }
 
+            // if value inherits from IModelSerializable, use the ModelSerializer serialize method otherwise use default JsonSerializer
+            // need to return an object that can be serialized by JsonSerializer.SerializeToUtf8Bytes and not BinaryData
+            if (value is IModelSerializable)
```

Comment:
I believe we need to ensure we'll get JSON when we serialize -- should we be checking for IJsonModelSerializable, or is it enough to specify Json in the ModelSerializerOptions to ensure this?